src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -189,6 +195,23 @@ private String getPayload(SinkRecord record) {
     return new String(rawJsonPayload, StandardCharsets.UTF_8);
   }
 
+  private String injectTimestampIfDataStreamAndAbsent(String payload, SinkRecord record) {
+    if (!config.isDataStream()) {
+      return payload;
+    }
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      JsonNode jsonNode = objectMapper.readTree(payload);
+      if (jsonNode.get("@timestamp").isNull()) {
+        ((ObjectNode)jsonNode).put("@timestamp", String.valueOf(record.timestamp()));
```

Comment:
```suggestion
        ((ObjectNode) jsonNode).put("@timestamp", String.valueOf(record.timestamp()));
```

Reply:
we should check the ES datatypes but we probably want to provide it as a number not as a string